src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.Baselines.Options.cs

```diff
@@ -60,6 +60,79 @@ public class MyClass2
         public async Task Configure_T_name_BinderOptions() =>
             await VerifyAgainstBaselineUsingFile("Configure_T_name_BinderOptions.generated.txt", GetConfigureSource(@""""", section, _ => { }"), extType: ExtensionClassType.ServiceCollection);
 
+        [Fact]
+        public async Task Configure_T_NamedParameters_OutOfOrder()
+        {
+            string source = """
+                    using System.Collections.Generic;
+                    using Microsoft.Extensions.Configuration;
+                    using Microsoft.Extensions.DependencyInjection;
+                    
+                    public class Program
+                    {
+                        public static void Main()
+                        {
+                            ConfigurationBuilder configurationBuilder = new();
+                            IConfiguration config = configurationBuilder.Build();
+                            IConfigurationSection section = config.GetSection("MySection");
+                            ServiceCollection services = new();
+
+                            OptionsConfigurationServiceCollectionExtensions.Configure<MyClass>(config: section, services: services);
```

Comment:
We need to (also) have separate tests for each overload, to validate that every overload is handled correctly. For these & for `ConfigurationBinder`.

Reply:
I'm surprised that no changes were needed for the `Options.ConfigurationExtensions` methods. Testing each one in isolation might reveal a gap.

Reply:
> I'm surprised that no changes were needed for the `Options.ConfigurationExtensions` methods. Testing each one in isolation might reveal a gap.

Not surprising to me, because those implementations did not use/reference arguments by index at all. Anyway, separated the test for each overload.